### PR TITLE
feat: SDK 2.0 Support

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
This PR enables support for 2.0 of the Sentry SDK. The changes are likely incompatible with Sentry SDK 1.x.

We are still in process of testing this PR, hence why it is marked as a draft. We should wait to merge until we verify that Sentry tests pass when run against this `pytest-sentry` version

Fixes GH-27

### CODE REVIEW

File: pytest_sentry.py [hunk 1]
```diff
@@ -94,52 +97,52 @@ def __init__(self, *args, **kwargs):
...
         item = itemgetter(*args, **kwargs)
-        hub = _resolve_hub_marker_value(item.get_closest_marker("sentry_client"))
+        scope = _resolve_scope_marker_value(item.get_closest_marker("sentry_client"))
 
-        if hub.get_integration(PytestIntegration) is None:
+        if scope.client.get_integration(PytestIntegration) is None:
             yield
         else:
-            with hub:
+            with use_isolation_scope(scope):  # TODO: What do we do with scope here?
```
Comment (untitaker) [inline pytest_sentry.py:114]: we should not fork the scope here. we need to set it as current scope and unset it at the end of the with-stmt. I think `use_scope` is more correct

specifically, `sentry_test_scope` should be usable from within tests to access the transaction created by the plugin, and modify tags. I think the tags might work but the assertion probably fails

```
def test_foo(sentry_test_scope):
    assert sentry_test_scope.span is not None
    sentry_test_scope.set_tag("foo", "bar")
```

Comment (szokeasaurusrex) [reply pytest_sentry.py]: Okay, I switched to `use_scope` and added a test with the span is not None assertion (the assertion failed with `use_isolation_scope`)

**Review by untitaker — APPROVED** [review]

File: README.rst [hunk 2]
```diff
@@ -125,12 +125,12 @@ you configured this plugin with. That's because ``pytest-sentry`` goes to
...
-``pytest-sentry`` exposes the ``sentry_test_hub`` fixture whose return value is
-the ``Hub`` being used to send events to Sentry. Use ``with sentry_test_hub:``
+``pytest-sentry`` exposes the ``sentry_test_scope`` fixture whose return value is
+the ``Scope`` being used to send events to Sentry. Use ``with sentry_test_scope:``
 to temporarily switch context. You can use this to set custom tags like so::
 
-    def test_foo(sentry_test_hub):
-        with sentry_test_hub:
+    def test_foo(sentry_test_scope):
+        with sentry_test_scope:
```
[the remark this answers is not in the document]
Comment (szokeasaurusrex) [reply README.rst:133]: Yeah, you are right. I should update this to `with use_scope(sentry_test_scope)`.

Do you know if we even use `sentry_test_hub` anywhere? I searched for it in the Sentry code base but did not find it

**Comment by szokeasaurusrex** [comment]
This commit enables support for 2.0 of the Sentry SDK. The changes are likely incompatible with Sentry SDK 1.x.

Fixes https://github.com/getsentry/pytest-sentry/issues/27

**Pull request merged**